File: android/app/src/main/kotlin/com/example/gen_confi/NativeCameraViewFactory.kt
package com.example.gen_confi

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class NativeCameraViewFactory(
    private val messenger: BinaryMessenger,
    private val eventChannel: EventChannel,
    private val onViewCreated: (NativeCameraView) -> Unit
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val view = NativeCameraView(context, messenger, eventChannel)
        onViewCreated(view)
        return view
    }
}

File: ios/Runner/NativeCameraViewFactory.swift
// FILE: ios/Runner/NativeCameraViewFactory.swift

import Flutter
import UIKit

class NativeCameraViewFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger
    private weak var plugin: SmartCapturePlugin?

    init(messenger: FlutterBinaryMessenger, plugin: SmartCapturePlugin) {
        self.messenger = messenger
        self.plugin = plugin
        super.init()
    }

    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        let view = NativeCameraView(
            frame: frame,
            viewIdentifier: viewId,
            arguments: args,
            binaryMessenger: messenger,
            plugin: plugin
        )
        // Register view with plugin so methods control it
        plugin?.activeCameraView = view
        return view
    }
    
    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
          return FlutterStandardMessageCodec.sharedInstance()
    }
}
